### PR TITLE
RFC: Add a toolchaine file for CMake in exported archive.

### LINKED ISSUE #10402: NuttX exported archive hard to use with CMake
The exported archive is hard to use with an external CMake application: one who wants to use it must write its own toolchain file dedicated to its project. Such toolchain file must be adapted every time NuttX build change (like switching to another target or version upgrade).

On the other hand, the exported archive contains a Make.defs that contains everything needed to generate a generic toolchain file so it should be possible to add a generic CMake toolchain file in the archive.

### PR DESCRIPTION
**Pull request by Siviuze**
## Summary
Add a CMake toolchain in the exported archive so that application can use it directly.
close #10402

### CODE REVIEW

**Comment by Siviuze** [comment]
Should I create then link an issue to this PR?

**Comment by xiaoxiang781216** [comment]
@Siviuze could you fix the warning reported by ci:
https://github.com/apache/nuttx/actions/runs/5973272863/job/16205220666?pr=10398

**Comment by Siviuze** [comment]
> @Siviuze could you fix the warning reported by ci: https://github.com/apache/nuttx/actions/runs/5973272863/job/16205220666?pr=10398

Fixed :)

File: tools/toolchainfile.cmake [hunk 1]
```diff
@@ -0,0 +1,36 @@
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_VERSION 1)
+
+include(${CMAKE_CURRENT_LIST_DIR}/target.cmake)
```
Comment (trns1997) [inline tools/toolchainfile.cmake:4]: small suggestion set nuttx path before include like this:
```
set(NUTTX_PATH ${CMAKE_CURRENT_LIST_DIR}/..)

include(${NUTTX_PATH}/scripts/target.cmake)
```

**Comment by Siviuze** [comment]
> > @Siviuze could you fix the warning reported by ci: https://github.com/apache/nuttx/actions/runs/5973272863/job/16205220666?pr=10398
> 
> Fixed :)

Fixed again : I messed up with my previous try.

File: tools/mkexport.sh [hunk 2]
```diff
@@ -265,8 +292,19 @@ if [ "X${USRONLY}" != "Xy" ]; then
...
+  echo "set(EXTRA_OBJS       \"${EXTRA_OBJS}\")"   >>"${EXPORTDIR}/scripts/target.cmake"
+  echo "set(HEAD_OBJ         \"${HEAD_OBJ}\")"     >>"${EXPORTDIR}/scripts/target.cmake"
+  echo "set(LDENDGROUP       \"${LDENDGROUP}\")"   >>"${EXPORTDIR}/scripts/target.cmake"
+  echo "set(LDFLAGS          \"${LDFLAGS}\")"      >>"${EXPORTDIR}/scripts/target.cmake"
+  echo "set(LDSTARTGROUP     \"${LDSTARTGROUP}\")" >>"${EXPORTDIR}/scripts/target.cmake"
 fi
 
+# Copy the CMake toolchain file
+
+cp "${TOPDIR}/tools/toolchainfile.cmake.export" "${EXPORTDIR}/scripts/toolchainfile.cmake"
```
[the remark this answers is not in the document]
Comment (Siviuze) [reply tools/mkexport.sh:306]: Standard suffix for CMake toolchain files is .cmake. I choose to add 'export' suffix in the tool folder to emphasis that the file is related to the export step and is not usable as-is (I didn't find a place for exported files). I can name it .cmake if you prefer?

[the remark this answers is not in the document]
Comment (Siviuze) [reply tools/mkexport.sh:306]: I took the name from buildroot project: I can rename it if you prefer.

P.S. My previous comment missed the point of your question so I deleted/updated it.

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply tools/mkexport.sh:306]: sorry, I mean toolchain.cmake.

[the remark this answers is not in the document]
Comment (Siviuze) [reply tools/mkexport.sh]: File renamed :)

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Pull request merged**